phoenix-core/src/main/java/org/apache/phoenix/expression/function/SignFunction.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.expression.function;
+
+import java.math.BigDecimal;
+import java.sql.SQLException;
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.expression.Expression;
+import org.apache.phoenix.parse.FunctionParseNode.Argument;
+import org.apache.phoenix.parse.FunctionParseNode.BuiltInFunction;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+import org.apache.phoenix.schema.types.PDecimal;
+import org.apache.phoenix.schema.types.PInteger;
+
+/**
+ *
+ * Base class for built-in SIGN function.
+ *
+ * @since 4.3.0
+ *
+ */
+@BuiltInFunction(name = SignFunction.NAME,
+                 args = {
+                        @Argument(allowedTypes={PDecimal.class})
+                        }
+                )
+public class SignFunction extends ScalarFunction {
```

Comment:
Hi James,

One question:

In PDataType.java,

  public abstract Object toObject(byte[] bytes, int offset, int length, PDataType actualType,
      SortOrder sortOrder, Integer maxLength, Integer scale);

What does "SortOrder sortOrder, Integer maxLength, Integer scale" mean here?

-------------------

I'd like to take the second solution, which introduces a new type PNumericType. 

So in PNumericType, we have an abstract function getSign, which is like function toObject in PDataType, which we implement the details in each subclass, PInteger, PFloat etc.

How about the solution above?

Thanks.